vm.go

```diff
@@ -1282,16 +1282,20 @@ func (_mul) exec(vm *vm) {
 
 	var result Value
 
-	if left, ok := assertInt64(left); ok {
-		if right, ok := assertInt64(right); ok {
+	if left, ok, leftIsInt64 := assertInt64(left); ok {
+		if right, ok, rightIsInt64 := assertInt64(right); ok {
 			if left == 0 && right == -1 || left == -1 && right == 0 {
 				result = _negativeZero
 				goto end
 			}
 			res := left * right
 			// check for overflow
 			if left == 0 || right == 0 || res/left == right {
-				result = intToValue(res)
+				if leftIsInt64 || rightIsInt64 {
+					result = int64ToValue(res)
+				} else {
+					result = intToValue(res)
+				}
```

Comment:
When multiplying an int and int64 this now makes it to where we would make the result an int64 which I believe is more in line with expectations 🤔

Reply:
Note that the same logic was implemented for other operations (below).

Reply:
If we run a test script to debug the current code, what's data type of the `result`?

Reply:
So the current code `result = intToValue(res)` returns the type as a valueInt unless it is smaller or bigger than min/max int values, in which case it gets cast to a valueFloat.

In my head I was assuming int was 32bits but I guess that is platform dependent 😅

Reply:
Perhaps I should only change the operations that change a single value (i.e. inc, dec, neg) since that makes more sense that we should maintain the type throughout, versus a little more nuance with multiply/mod/divide

Reply:
> Perhaps I should only change the operations that change a single value (i.e. inc, dec, neg) since that makes more sense that we should maintain the type throughout, versus a little more nuance with multiply/mod/divide

that makes sense to me.